### PR TITLE
Refractor Initializer

### PR DESCRIPTION
**Pull request by souravbaner-da**
Refractor Initializer

### CODE REVIEW

File: src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala [hunk 1]
```diff
@@ -566,6 +160,9 @@ object Initializer extends SparkSessionWrapper {
   private def initConfigState(debugFlag: Boolean,organizationID: Option[String],apiUrl: Option[String]): Config = {
     logger.log(Level.INFO, "Initializing Config")
     val config = new Config()
+    val orgId = if (config.isLocalTesting) System.getenv("ORGID") else {
+      getOrgId
+    }
```
Comment (GeekSheikh) [inline src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala:165]: let's remove this reference as isLocalTesting is deprecated. Let's just continue to use the `getOrgId` function as we are below.

**Review by GeekSheikh — CHANGES_REQUESTED** [review]
cursory analysis looks good. Please re-submit for review after @sriram251-code review is complete and requested PR Proof notebooks are completed and vetted.

**Comment by souravbaner-da** [comment]
Close #677

File: src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala [hunk 2]
```diff
@@ -296,6 +48,10 @@ class Initializer(config: Config) extends SparkSessionWrapper {
     val rawParams = mapper.readValue[OverwatchParams](overwatchArgs)
     config.setInputConfig(rawParams)
 
+    // Now that the input parameters have been parsed -- set them in the config
+    config.setInputConfig(rawParams)
```
[the remark this answers is not in the document]
Comment (souravbaner-da) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala:52]: @sriram251-code Done

File: src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala [hunk 3]
```diff
@@ -363,193 +103,47 @@ class Initializer(config: Config) extends SparkSessionWrapper {
     config.setPrimordialDateString(rawParams.primordialDateString)
 
     // Audit logs are required and paramount to Overwatch delivery -- they must be present and valid
-    if (!disableValidations) {
-      config.setAuditLogConfig(validateAuditLogConfigs(auditLogConfig))
-    } else {
-      config.setAuditLogConfig(quickBuildAuditLogConfig(auditLogConfig))
-    }
-
```
[the remark this answers is not in the document]
Comment (souravbaner-da) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala:371]: @sriram251-code  this has been moved to InitializerFunctions.scala

File: src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala [hunk 4]
```diff
@@ -363,193 +103,47 @@ class Initializer(config: Config) extends SparkSessionWrapper {
...
 
-    // must happen AFTER data target validation
-    if (!disableValidations) { // temp working dir is not necessary for disabled validations as pipelines cannot be
-      // executed without validations
-      prepAndSetTempWorkingDir(rawParams.tempWorkingDir, config.etlDataPathPrefix)
-    }
+    // must happen AFTER data target is set validation since the etlDataPrefix must already be set in the config
+    /** Validate and set Temp Working Dir */
+    val rawTempWorkingDir = rawParams.tempWorkingDir
+    val validatedTempWorkingDir = Init.validateTempWorkingDir(rawTempWorkingDir)
```
[the remark this answers is not in the document]
Comment (souravbaner-da) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala:123]: @sriram251-code Done. Changed to prepTempWorkingDir

File: src/main/scala/com/databricks/labs/overwatch/pipeline/InitializerFunctions.scala [hunk 5]
```diff
@@ -1,30 +1,498 @@
...
+    /**
+     * validate the tokenSecret provided is acceptable
+     * @param tokenSecret OW TokenSecret to be validated
+     * @param disableValidations Whether or not the Initializer validations are disabled
+     * @return
+     */
+    def validateTokenSecret(tokenSecret: Option[TokenSecret]): Option[TokenSecret] = {
+      // validate token secret requirements
+      // TODO - Validate if token has access to necessary assets. Warn/Fail if not
+      if (tokenSecret.nonEmpty && !disableValidations && !config.isLocalTesting) {
```
[the remark this answers is not in the document]
Comment (souravbaner-da) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/InitializerFunctions.scala:110]: @sriram251-code  Done

File: src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala [hunk 6]
```diff
@@ -558,6 +152,8 @@ object Initializer extends SparkSessionWrapper {
   envInit()
 
   def getOrgId: String = {
+
+    print("tags are ", dbutils.notebook.getContext.tags)
     if (dbutils.notebook.getContext.tags("orgId") == "0") {
```
[the remark this answers is not in the document]
Comment (souravbaner-da) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala:157]: Already done in another issue https://github.com/databrickslabs/overwatch/issues/115

File: src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala [hunk 7]
```diff
@@ -558,6 +152,8 @@ object Initializer extends SparkSessionWrapper {
   envInit()
 
   def getOrgId: String = {
+
+    print("tags are ", dbutils.notebook.getContext.tags)
```
[the remark this answers is not in the document]
Comment (souravbaner-da) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala:156]: @sriram251-code  done

File: src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala [hunk 8]
```diff
@@ -363,193 +103,47 @@ class Initializer(config: Config) extends SparkSessionWrapper {
...
-      prepAndSetTempWorkingDir(rawParams.tempWorkingDir, config.etlDataPathPrefix)
-    }
+    // must happen AFTER data target is set validation since the etlDataPrefix must already be set in the config
+    /** Validate and set Temp Working Dir */
+    val rawTempWorkingDir = rawParams.tempWorkingDir
+    val validatedTempWorkingDir = Init.validateTempWorkingDir(rawTempWorkingDir)
+    config.setTempWorkingDir(validatedTempWorkingDir)
 
+    /** Set Max Days */
     config.setMaxDays(rawParams.maxDaysToLoad)
```
[the remark this answers is not in the document]
Comment (souravbaner-da) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala:127]: Could you please let me know the logic if maxDaysToLoad would be 0 or negative number?

[the remark this answers is not in the document]
Comment (souravbaner-da) [reply src/main/scala/com/databricks/labs/overwatch/pipeline/Initializer.scala]: As per discussion done the changes

**Review by sriram251-code — APPROVED** [review]

**Review by GeekSheikh — APPROVED** [review]

**Pull request merged**